Branch-SDK/Branch-SDK/ContentDiscovery/BNCContentDiscoveryManager.m

```diff
@@ -0,0 +1,138 @@
+}
+
+
+#pragma mark - Content Indexing
+
+- (void)indexContentWithTitle:(NSString *)title description:(NSString *)description {
+    [self indexContentWithTitle:title description:description publiclyIndexable:NO type:(NSString *)kUTTypeImage thumbnailUrl:nil keywords:nil userInfo:nil callback:NULL];
+}
+
+- (void)indexContentWithTitle:(NSString *)title description:(NSString *)description callback:(callbackWithUrl)callback {
+    [self indexContentWithTitle:title description:description publiclyIndexable:NO type:(NSString *)kUTTypeImage thumbnailUrl:nil keywords:nil userInfo:nil callback:callback];
+}
+
+- (void)indexContentWithTitle:(NSString *)title description:(NSString *)description publiclyIndexable:(BOOL)publiclyIndexable callback:(callbackWithUrl)callback {
+    [self indexContentWithTitle:title description:description publiclyIndexable:publiclyIndexable type:(NSString *)kUTTypeImage thumbnailUrl:nil keywords:nil userInfo:nil callback:callback];
+}
+
+- (void)indexContentWithTitle:(NSString *)title description:(NSString *)description publiclyIndexable:(BOOL)publiclyIndexable type:(NSString *)type callback:(callbackWithUrl)callback {
+    [self indexContentWithTitle:title description:description publiclyIndexable:publiclyIndexable type:type thumbnailUrl:nil keywords:nil userInfo:nil callback:callback];
+}
+
+- (void)indexContentWithTitle:(NSString *)title description:(NSString *)description publiclyIndexable:(BOOL)publiclyIndexable type:(NSString *)type thumbnailUrl:(NSURL *)thumbnailUrl callback:(callbackWithUrl)callback {
+    [self indexContentWithTitle:title description:description publiclyIndexable:publiclyIndexable type:type thumbnailUrl:thumbnailUrl keywords:nil userInfo:nil callback:callback];
+}
+
+- (void)indexContentWithTitle:(NSString *)title description:(NSString *)description publiclyIndexable:(BOOL)publiclyIndexable type:(NSString *)type thumbnailUrl:(NSURL *)thumbnailUrl keywords:(NSSet *)keywords callback:(callbackWithUrl)callback {
+    [self indexContentWithTitle:title description:description publiclyIndexable:publiclyIndexable type:type thumbnailUrl:thumbnailUrl keywords:keywords userInfo:nil callback:callback];
+}
+
+- (void)indexContentWithTitle:(NSString *)title description:(NSString *)description publiclyIndexable:(BOOL)publiclyIndexable type:(NSString *)type thumbnailUrl:(NSURL *)thumbnailUrl keywords:(NSSet *)keywords userInfo:(NSDictionary *)userInfo callback:(callbackWithUrl)callback {
+    if ([BNCSystemObserver getOSVersion].integerValue < 9) {
+        if (callback) {
+            callback(nil, [NSError errorWithDomain:BNCErrorDomain code:BNCVersionError userInfo:@{ NSLocalizedDescriptionKey: @"Cannot use CoreSpotlight indexing service prior to iOS 9" }]);
+        }
+        return;
+    }
+
+    #if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 90000
+        [[Branch getInstance] getSpotlightUrlWithParams:userInfo callback:^(NSDictionary *data, NSError *urlError) {
```

Comment:
Is there a reason we don't pass up title, description, etc? Perhaps as "+title", "+description", etc. Or perhaps I'm overlooking something.